scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1120,12 +1121,34 @@
     (.setMetadata pod metadata)
     (.setSpec pod pod-spec)
 
-    ; We want to allow synthetic pods to have a non-default (typically 0) termination grace period,
-    ; so that they can be deleted quickly to free up space on nodes for real job pods. The default
-    ; grace period of 30 seconds can cause real job pods to be deemed unschedulable and fail.
     (when (synthetic-pod? pod-name)
+      ; We want to allow synthetic pods to have a non-default (typically 0) termination grace period,
+      ; so that they can be deleted quickly to free up space on nodes for real job pods. The default
+      ; grace period of 30 seconds can cause real job pods to be deemed unschedulable and fail.
       (when-let [{:keys [synthetic-pod-termination-grace-period-seconds]} (config/kubernetes)]
-        (.setTerminationGracePeriodSeconds pod-spec synthetic-pod-termination-grace-period-seconds)))
+        (.setTerminationGracePeriodSeconds pod-spec synthetic-pod-termination-grace-period-seconds))
+
+      ; We want to allow synthetic pods to be repelled from certain nodes via inter-pod (anti-)affinity
+      ; (https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity)
+      (let [{:keys [synthetic-pod-anti-affinity-pod-label-key
+                    synthetic-pod-anti-affinity-pod-label-value]}
+            (config/kubernetes)]
+        (when (and synthetic-pod-anti-affinity-pod-label-key
+                   synthetic-pod-anti-affinity-pod-label-value)
+          ; The synthetic pod pod-spec may or may not already have an affinity defined
+          ; (see pod-hostnames-to-avoid above), so we need to allow for either case
+          (let [affinity (or (.getAffinity pod-spec) (V1Affinity.))
```

Comment:
I'd rephrase this as: 
""" If the synthetic pod already has an affinity (see ....)  we add to it, otherwise, we create a new one """

Reply:
Agreed, done